### ISSUE TITLE
optional parameters

### ISSUE THREAD

**Issue by JuanMarchetto** [seriousben/embeddable-react-widget#119]
hello,i think this is a great idea, and i want to suggest an small change that increment the funcionality: contemplate the use of parameter in the inicialization function of the widget, so for example you can pass a parent of wich the widget will be children, and also pass parameters to the main component so you can set some options. This will be an optional parameter so if in unset the functionality will be the same as now, the only changes neded will be in the embeddable.widget.js:

```

import React from 'react';
import ReactDOM from 'react-dom';
import Widget from '../components/widget';
import '../../vendor/cleanslate.css';

export default class EmbeddableWidget {
  static el;

  static mount(params=null) {
    const component = <Widget params={params}/>;

    function doRender() {
      if (EmbeddableWidget.el) {
        throw new Error('EmbeddableWidget is already mounted, unmount first');
      }
      const el = document.createElement('div');
      el.setAttribute('class', 'cleanslate');
      if (params && params.parent) {
        document.querySelector(params.parent).appendChild(el);
      }else{
        document.body.appendChild(el);
      }
      ReactDOM.render(
        component,
        el,
      );
      EmbeddableWidget.el = el;
    }
    if (document.readyState === 'complete') {
      doRender();
    } else {
      window.addEventListener('load', () => {
        doRender();
      });
    }
  }

  static unmount() {
    if (!EmbeddableWidget.el) {
      throw new Error('EmbeddableWidget is not mounted, mount first');
    }
    ReactDOM.unmountComponentAtNode(EmbeddableWidget.el);
    EmbeddableWidget.el.parentNode.removeChild(EmbeddableWidget.el);
    EmbeddableWidget.el = null;
  }
}

```

with this change you can initializate de widget this way:

`EmbeddableWidget.mount({parent:'#mountingPoint'});`

and the component will be a child of the element with mountingPoint id and the parameters are passed tu de widget component so you can set some option if you will

or if you do the traditiona initialization:

`EmbeddableWidget.mount();`

will be child of document

**Comment by seriousben** [comment]
This is a great idea. That would be an amazing contribution if you had some time.

**Comment by JuanMarchetto** [comment]
i already write the code, i need Permission to do the pull request

**Comment by seriousben** [comment]
@JuanMarchetto following this guide should allow you to Fork this repo and create a Pull Request from your own fork.

https://github.com/firstcontributions/first-contributions

**Comment by fostergn** [comment]
Hey I didn't check the other PRs and needed similar functionality for optional properties https://github.com/seriousben/embeddable-react-widget/pull/123. This is passing tests, but I'll add the mounting point to my PR in case that would be easier to merge

**Comment by seriousben** [comment]
Fixed in #123
